In [3]:
%matplotlib inline
import pandas as pd

In [2]:
from IPython.core.display import HTML
css = open('style-table.css').read() + open('style-notebook.css').read()
HTML('<style>{}</style>'.format(css))

In [5]:
titles = pd.read_csv('C:/Machine Learning/DAY16/titles.csv')
titles

title  year
0                        Tasveer Mere Sanam  1996
1                                  Only You  1994
2                      El pueblo del terror  1970
3                                   Machine  2007
4                                      MARy  2008
...                                     ...   ...
244909  Black Butterfly in a Colorful World  2018
244910        Hua fei hua wu chun man cheng  1980
244911                 Nippon dabi katsukyu  1970
244912        Under Siege 2: Dark Territory  1995
244913            She Must Be Seeing Things  1987

[244914 rows x 2 columns]

In [ ]:
tit


### How many movies are listed in the titles dataframe?

In [11]:
titles.count()

title    244914
year     244914
dtype: int64

There are totally 244914 movies listed

### What are the earliest two films listed in the titles dataframe?

In [13]:
EarliestFilms=titles.sort_values('year',ascending=True)
print(EarliestFilms.head(2))

                     title  year
193222          Miss Jerry  1894
51388   The Startled Lover  1898


In [ ]:
##Miss Jerry ,The Startled Lover are two films released earlier

### How many movies have the title "Hamlet"?

In [24]:
titles[titles['title'] == "Hamlet"].count()

title    20
year     20
dtype: int64

### How many movies are titled "North by Northwest"?

In [22]:
titles[titles['title'] == "North by Northwest"].count()


title    1
year     1
dtype: int64

### When was the first movie titled "Hamlet" made?

In [25]:
Hamlet=titles[titles['title'] == "Hamlet"]
Hamlet.sort_values('year',ascending=True).head(1)

title  year
52521  Hamlet  1910

In [ ]:
The First Hamlet movie was made in 1910

### List all of the "Treasure Island" movies from earliest to most recent.

In [28]:
Treasure_Island=titles[titles['title'] == "Treasure Island"]
print(Treasure_Island.sort_values('year',ascending=True))

                  title  year
55686   Treasure Island  1918
196514  Treasure Island  1920
141822  Treasure Island  1934
77354   Treasure Island  1950
65051   Treasure Island  1972
37238   Treasure Island  1973
173051  Treasure Island  1985
27076   Treasure Island  1999


### How many movies were made in the year 1950?

In [30]:
titles[titles['year'] == 1950].count()

title    1113
year     1113
dtype: int64

In [ ]:
There are 1113 movies made in year 1950

### How many movies were made in the year 1960?

In [31]:
titles[titles['year'] == 1960].count()

title    1549
year     1549
dtype: int64

In [ ]:
## 1549 movies were made in year 1960

### How many movies were made from 1950 through 1959?

In [35]:
titles[(titles['year'] >= 1950) & (titles['year'] <= 1959)]['title'].count()

12934

##12934 movies were made in year 1950 to 1959

### In what years has a movie titled "Batman" been released?

In [6]:
cast=pd.read_csv('C:/Machine Learning/DAY16/cast.csv')
cast.head(3)
listt1 = titles[titles['title'] == "Batman"]['year'].tolist()
listt1.sort()
print("The movie titled 'Batman' has been released in the years", *listt1)

The movie titled 'Batman' has been released in the years 1943 1989


### How many roles were there in the movie "Inception"?

In [37]:
cast[cast['title'] == 'Inception']['type'].count()

78

### How many roles in the movie "Inception" are NOT ranked by an "n" value?

In [38]:
cast[cast['title'] == 'Inception']['n'].isnull().sum()

27

### But how many roles in the movie "Inception" did receive an "n" value?

In [39]:
cast[cast['title'] == 'Inception']['n'].notnull().sum()

51

### Display the cast of "North by Northwest" in their correct "n"-value order, ignoring roles that did not earn a numeric "n" value.

In [40]:
cast1 = cast[cast.n.notnull()].sort_values(by=['n'])
l = cast1[cast1['title'] == 'North by Northwest'][['name','n']].values.tolist()
print("The cast of 'North by Northwest' in their correct 'n'-value order,  ignoring roles that did not earn a numeric 'n' value:")
nbn = pd.DataFrame(l, columns = ['Name', 'n']) 
nbn

The cast of 'North by Northwest' in their correct 'n'-value order,  ignoring roles that did not earn a numeric 'n' value:


Name     n
0             Cary Grant   1.0
1        Eva Marie Saint   2.0
2            James Mason   3.0
3    Jessie Royce Landis   4.0
4         Leo G. Carroll   5.0
5   Josephine Hutchinson   6.0
6            Philip Ober   7.0
7          Martin Landau   8.0
8          Adam Williams   9.0
9           Edward Platt  10.0
10     Robert Ellenstein  11.0
11          Les Tremayne  12.0
12       Philip Coolidge  13.0
13         Patrick McVey  14.0
14          Edward Binns  15.0
15             Ken Lynch  16.0

### Display the entire cast, in "n"-order, of the 1972 film "Sleuth".

In [41]:
cast2 = cast[cast.n.notnull()].sort_values(by=['n'])
print('The entire cast, in "n"-order, of the 1972 film "Sleuth":')
cast2 = cast2.loc[(cast2['title']=='Sleuth') & (cast2['year']==1972)][['name','n']].reset_index(drop=True)
cast2

The entire cast, in "n"-order, of the 1972 film "Sleuth":


name    n
0    Laurence Olivier  1.0
1       Michael Caine  2.0
2      Alec Cawthorne  3.0
3  John (II) Matthews  4.0
4  Eve (III) Channing  5.0
5        Teddy Martin  6.0

### Now display the entire cast, in "n"-order, of the 2007 version of "Sleuth".

In [42]:
cast3 = cast[cast.n.notnull()].sort_values(by=['n'])
print('The entire cast, in "n"-order, of the 2007 film "Sleuth":')
cast3 = cast3.loc[(cast3['title']=='Sleuth') & (cast3['year']==2007)][['name','n']].reset_index(drop=True)
cast3

The entire cast, in "n"-order, of the 2007 film "Sleuth":


name    n
0  Michael Caine  1.0
1       Jude Law  2.0
2  Harold Pinter  3.0

### How many roles were credited in the silent 1921 version of Hamlet?

In [43]:
cast4 = cast.loc[(cast['title']=='Hamlet') & (cast['year']==1921)]
cast4['character'].count()

9

### How many roles were credited in Branagh’s 1996 Hamlet?

In [44]:
cast5 = cast.loc[(cast['title']=='Hamlet') & (cast['year']==1996)]
cast5['character'].count()

54

### How many "Hamlet" roles have been listed in all film credits through history?

In [45]:
cast6 = cast[cast['title'] == 'Hamlet']
cast6['character'].count()

342

### How many people have played an "Ophelia"?

In [46]:
cast[cast['character'] == 'Ophelia']['name'].count()

117

### How many people have played a role called "The Dude"?

In [47]:

cast[cast['character'] == 'The Dude']['name'].count()

19

### How many people have played a role called "The Stranger"?

In [48]:
cast[cast['character'] == 'The Stranger']['name'].count()

212

### How many roles has Sidney Poitier played throughout his career?

In [49]:
cast[cast['name'] == "Sidney Poitier"]['character'].count()

43

### How many roles has Judi Dench played?

In [50]:

cast[cast['name'] == "Judi Dench"]['character'].count()

55

### List the supporting roles (having n=2) played by Cary Grant in the 1940s, in order by year.

In [51]:
cast6 = cast[(cast['name'] == "Cary Grant") & (cast['n'] == 2) & (cast['year'] >= 1940) & (cast['year'] < 1950)]
li = cast6[['character','year']].values.tolist()
print("Following are the supporting roles played by Cary Grant in the 1940s:\n")
cg = pd.DataFrame(li, columns = ['Supporting Role', 'Year'])
cg

Following are the supporting roles played by Cary Grant in the 1940s:



Supporting Role  Year
0      Nick Arden  1940
1     Roger Adams  1941

### List the leading roles that Cary Grant played in the 1940s in order by year.

In [7]:
cast7 = cast[(cast['name'] == "Cary Grant") & (cast['n'] == 1) & (cast['year'] >= 1940) & (cast['year'] < 1950)].sort_values(by = ['year'])
lis= cast7[['character','year']].values.tolist()
print("Following are the leading roles played by Cary Grant in the 1940s:\n")
cg_lead = pd.DataFrame(lis, columns = ['Leading Role', 'Year'])
cg_lead

Following are the leading roles played by Cary Grant in the 1940s:



Leading Role  Year
0                            Matt Howard  1940
1                           Walter Burns  1940
2                     C. K. Dexter Haven  1940
3                       Johnnie Aysgarth  1941
4                           Leopold Dilg  1942
5                  Patrick 'Pat' O'Toole  1942
6                          Capt. Cassidy  1943
7   Joe Adams -posing as Joe Bascopolous  1943
8                            Jerry Flynn  1944
9                      Mortimer Brewster  1944
10                            Ernie Mott  1944
11                           Cole Porter  1946
12                                Devlin  1946
13                           Dick Nugent  1947
14                                Dudley  1947
15                         Jim Blandings  1948
16                     Dr. Madison Brown  1948
17                 Captain Henri Rochard  1949

### How many roles were available for actors in the 1950s?

In [8]:
cast[(cast['year'] >= 1950) & (cast['year'] < 1960) & (cast['type'] == 'actor')]['character'].count()

157738

### How many roles were available for actresses in the 1950s?

In [9]:
cast[(cast['year'] >= 1950) & (cast['year'] < 1960) & (cast['type'] == 'actress')]['character'].count()

57733

### How many leading roles (n=1) were available from the beginning of film history through 1980?

In [10]:
cast[(cast['year'] >= 1894) & (cast['year'] <= 1980) & (cast['n'] == 1)]['character'].count()

65140

### How many non-leading roles were available through from the beginning of film history through 1980?

In [11]:
cast[(cast['year'] >= 1894) & (cast['year'] <= 1980) & (cast['n'] > 1)]['character'].count()

669320

### How many roles through 1980 were minor enough that they did not warrant a numeric "n" rank?

In [12]:
cast[(cast['year'] >= 1894) & (cast['year'] <= 1980)]['n'].isnull().sum()

448347